contrib/email-notifications/auter-notify.module

```diff
@@ -0,0 +1,218 @@
+#!/bin/bash
+#
+# Send an email notification with the status of the auter prepare stage.
+#
+# auter is a yum-cron type package which implements automatic updates on an
+# individual server with features such as predownloading packages and reboots.
+#
+#
+# Copyright 2016 Rackspace, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+# this file except in compliance with the License.  You may obtain a copy of the
+# License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations under the License.
+#
+
+function main() {
+  function logit() {
+    # If running on a tty, or the --stdout option is provided, print to screen
+    ( tty -s || [[ -n "$STDOUT" ]] ) && echo "$1"
+    logger -p INFO -t auter "$1"
+  }
+
+  # Setup
+  local STAGE="$1"
+  if [[ -f "${SCRIPTDIR}/auter-notify.conf" ]]; then
+    source "${SCRIPTDIR}/auter-notify.conf"
+  else
+    logit "ERROR: auter-notify.conf is missing, skipping $0"
+  fi
+
+  local MAIL_CC
+  MAIL_CC="-c $(sed 's/ / -c /g' <<< "$MAIL_CC")"
+
+  if [[ "$STAGE" == "post-prep" ]] || [[ "$STAGE" == "pre-apply" ]]; then
+    local OUTPUT_FILE="${DATADIR}/last-prep-output-${CONFIGSET}"
+  elif [[ "$STAGE" == "post-apply" ]]; then
+    local OUTPUT_FILE="${DATADIR}/last-apply-output-${CONFIGSET}"
+  fi
+
+  # Sanity check some of the variables we can't live without
+  [[ -z "$STAGE" ]] &&  logit "ERROR: auter-notify requires a phase argument, skipping $0 script" && exit 0
+  [[ -z "$MAIL_TO" ]] && logit "ERROR: \$MAIL_TO is undefined, skipping $0 script" && exit 0
+  [[ ! -x $(which mailx) ]] && logit "ERROR: mailx is not installed, skipping $0 script" && exit 0
+
+  function get_prep_packages() {
+    awk --posix '/={70,}/,/Total download size/' "$OUTPUT_FILE"
+  }
+  function get_apply_packages() {
+    awk '/Updated:/,/Complete!/' "$OUTPUT_FILE"
+  }
+  function get_error_message() {
+    awk '/Error Downloading Packages:|Error Summary/,0' "$OUTPUT_FILE"
+  }
+  function get_stage_result() {
+    # Last $STAGE result, either FAIL or SUCCESS
+    if grep -E 'Error Downloading Packages|Error Summary' "$OUTPUT_FILE" &>/dev/null; then
+      local ret="FAIL"
+    fi
+    echo "${ret:-SUCCESS}"
+  }
+  function get_datetime() {
+    date -d "${1:-now}" "+%T %Z"
+  }
+  function run_custom_scripts {
```

Comment:
No issue here, this is a nice idea. Just make sure you document this in the README.md, especially the fact that this needs to be executable. 

You may also want to add an else clause to at lease log that some files have been skipped.